integration-cli/docker_cli_daemon_test.go

```diff
@@ -1889,7 +1889,9 @@ func (s *DockerDaemonSuite) TestDaemonNoSpaceleftOnDeviceError(c *check.C) {
 	c.Assert(err, check.IsNil)
 
 	// pull a repository large enough to fill the mount point
-	out, err := s.d.Cmd("pull", "registry:2")
+	// use a frozen image, because it's already locally available
+	imageName := fmt.Sprintf("%v/dockercli/debian:jessie", privateRegistryURL)
+	out, err := s.d.Cmd("pull", imageName)
 	c.Assert(out, check.Not(check.Equals), 1, check.Commentf("no space left on device"))
```

Comment:
I'm ok with this change as long as we change the test to verify that the error we get back is the error we're looking for. When I run it I see this:
```
+ go test -check.f TestDaemonNoSpaceleftOnDeviceError -test.timeout=180m -check.v github.com/docker/docker/integration-cli
INFO: Testing against a local daemon
out: Pulling repository 127.0.0.1:5000/dockercli/debian
Error while pulling image: Get http://127.0.0.1:5000/v1/repositories/dockercli/debian/images: dial tcp 127.0.0.1:5000: getsockopt: connection refused

err:exit status 1
PASS: docker_cli_daemon_test.go:1868: DockerDaemonSuite.TestDaemonNoSpaceleftOnDeviceError	0.689s
OK: 1 passed
PASS
```
I added a printf for "out" and "err".  Notice the error isn't what's expected.
Let's add a check to make sure the error message includes "no space left on device" - even if we don't change where we pull it from - since checking for an exit code of 1 is clearly not sufficient.